code/modules/projectiles/guns/ballistic/automatic.dm

```diff
@@ -164,11 +164,8 @@
 	underbarrel = new /obj/item/gun/ballistic/revolver/grenadelauncher/unrestricted(src)
 	update_icon()
 
-/obj/item/gun/ballistic/automatic/m90/afterattack(atom/target, mob/living/user, flag, params)
-	if(select == 2)
-		underbarrel.afterattack(target, user, flag, params)
-	else
-		return ..()
+/obj/item/gun/ballistic/automatic/m90/afterattack_secondary(atom/target, mob/living/user, flag, params)
+	underbarrel.afterattack(target, user, flag, params)
```

Comment:
This shouldn't return null. You'll find that it'll runtime when you right click someone adjacent to you.

Return `SECONDARY_ATTACK_CONTINUE_CHAIN` instead.